From f8bd48be884fb26bfa6a400809875cffdf59f656 Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Mon, 11 Aug 2014 12:26:34 -0400
Subject: [mfd] lpc_ich: Only configure watchdog or GPIO when present

Message-id: <1407759999-16143-4-git-send-email-prarit@redhat.com>
Patchwork-id: 87599
O-Subject: [RHEL7.1 PATCH BZ 1084134 3/8] mfd: lpc_ich: Only configure watchdog or GPIO when present
Bugzilla: 1084134
RH-Acked-by: Jarod Wilson <jarod@redhat.com>
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1084134

commit f0776b8ce03ceb638c51b62f324844c71c446600
Author: Peter Tyser <ptyser@xes-inc.com>
Date:   Mon Mar 10 16:34:52 2014 -0500

    mfd: lpc_ich: Only configure watchdog or GPIO when present

    Some chipsets don't currently have GPIO support enabled.  For these
    chipsets don't go through the process of initializing the GPIO region.

    Make the same change for the watchdog initialization for chipsets which
    may not enable the WDT in the future.

    Signed-off-by: Peter Tyser <ptyser@xes-inc.com>
    Tested-by: Rajat Jain <rajatjain@juniper.net>
    Reviewed-by: Guenter Roeck <linux@roeck-us.net>
    Signed-off-by: Lee Jones <lee.jones@linaro.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/mfd/lpc_ich.c b/drivers/mfd/lpc_ich.c
index 165bb0b..4f59b8d 100644
--- a/drivers/mfd/lpc_ich.c
+++ b/drivers/mfd/lpc_ich.c
@@ -968,13 +968,17 @@ static int lpc_ich_probe(struct pci_dev *dev,
 
  pci_set_drvdata(dev, priv);
 
- ret = lpc_ich_init_wdt(dev);
- if (!ret)
-  cell_added = true;
+ if (lpc_chipset_info[priv->chipset].iTCO_version) {
+  ret = lpc_ich_init_wdt(dev);
+  if (!ret)
+   cell_added = true;
+ }
 
- ret = lpc_ich_init_gpio(dev);
- if (!ret)
-  cell_added = true;
+ if (lpc_chipset_info[priv->chipset].gpio_version) {
+  ret = lpc_ich_init_gpio(dev);
+  if (!ret)
+   cell_added = true;
+ }
 
  /*
   * We only care if at least one or none of the cells registered
-- 
1.7.1